indico/modules/events/registration/fields/simple.py

```diff
@@ -388,3 +389,24 @@ def _indico_email(value):
                 raise ValidationError(_('Invalid email address'))
 
         return _indico_email
+
+
+class PictureField(FileField):
+    name = 'picture'
+    setup_schema_fields = {
+        'min_picture_size': fields.Integer(load_default=0, validate=validate.Range(0, 1200)),
+    }
+
+    def get_validators(self, existing_registration):
+        def _picture_size_and_type(value):
+            min_picture_size = self.form_item.data.get('min_picture_size') or None
+            file = File.query.filter(File.uuid == value, ~File.claimed).first() if value else None
```

Comment:
```
if not file:
    return
```

instead of repeating the `file and` below all the time.

even better:

```python
if not value:
    return
file = ...
if not file:
    raise ValidationError('Invalid file')
```

This should not happen (hence no need for i18n on that error), but it's a bit cleaner than just silently not doing anything if the UUId does not point to a valid file.

Reply:
ok thanks, changed it to not check the file multiple times